File: Python/factorial.py
# Python program to calculate factorial of the entered number.

n = int(input('Enter a number : '))
fact = 1

if(n<0):
    print('Enter a non-negative number')
elif( n==0 or n==1):
    print('Factorial of '+str(n)+' is : '+ str(1))
else:
    for i in range(1, n+1):
        fact = fact * i
    print('Factorial of '+str(n)+' is : '+str(fact))


# Printing result using f string or string literal
# print(f'Factorial of {n} is : {fact}')

File: C/factorial.c
// C program to calculate factorial of a given entered number.

#include<stdio.h>

int main(){
    int n, i, fact = 1;

    printf("Enter a number : ");
    scanf("%d", &n);

    if(n == 0 || n == 1){
        printf("Factorial of %d is 1", n);
    }
    else if(n < 0){
        printf("You entered a negative number, enter a positive number");
    }
    else{
        for(i = 1; i <= n; i++){
            fact = fact * i;
        }
    printf("Factorial of %d is %d", n, fact);
    }

    return 0;
}
